edb/server/protocol/auth_ext/_static/interactions.js

```diff
@@ -0,0 +1,133 @@
+document.addEventListener("DOMContentLoaded", () => {
+  const sliderContainer = document.getElementById("slider-container");
+
+  if (!sliderContainer) {
+    return;
+  }
+
+  const tabsContainer = document.getElementById("email-provider-tabs");
+  if (tabsContainer) {
+    const tabButtons = tabsContainer.children;
+    for (let i = 0; i < tabButtons.length; i++) {
+      tabButtons[i].addEventListener("click", () => {
+        setActiveClass(tabButtons, i);
+        moveSliderToIndex(sliderContainer, i);
+      });
+    }
+  } else {
+    const form = document.getElementById("email-factor");
+    let mainFormAction = form.action;
+
+    const hiddenInputs = Array.from(
+      form.querySelectorAll("input[type=hidden]")
+    ).filter((input) => !!input.dataset.secondaryValue);
+    const hiddenInputValues = hiddenInputs.map((input) => input.value);
+
+    console.log(hiddenInputs, hiddenInputValues);
+
+    if (!sliderContainer.children[0].classList.contains("active")) {
+      form.action = form.dataset.secondaryAction;
+      setInputValues(hiddenInputs);
+    }
+
+    document
+      .getElementById("show-password-form")
+      .addEventListener("click", () => {
+        moveSliderToIndex(sliderContainer, 1);
+
+        form.action = form.dataset.secondaryAction;
```

Comment:
Does that work if you don't submit the form using the button, like pressing the Enter key? If you have password + magic link, then both buttons exist, so I'm not sure which action would be used?

Reply:
I can't seem to get that to work. By default it always uses the `formaction` on the first button, even when it's hidden, I guess because it's only visually hidden. Trying to toggle disabled on the hidden buttons, just seems to act like there's no action at all. So it looks like the `formaction` on the hidden buttons would need to be removed when they're hidden, which I feel just brings us full circle.

Reply:
Ahh, yeah, if a button has `type="button"` it won't submit at all, which makes sense here. We _could_ use some kind of layout hack to put the buttons in a different order than document-order, but yeah I don't care enough to keep fighting this 😅 

Maybe another approach would be to switch to `fetch` -> `window.location` redirect at some point. My only real issue with the current approach is that it's a little hard to follow the logic unless you read everything altogether and even then I think I'd be prone to introducing subtle bugs here.

Not a blocker, let's not bikeshed on this too much!